### ISSUE TITLE
Swapped Values for Precision and Recall

### ISSUE THREAD

**Issue by FatemehEsfahani** [NRCan/geo-deep-learning#592]
**Describe the bug**
Precision and recall values in [Line 132](https://github.com/NRCan/geo-deep-learning/blob/develop/utils/metrics.py#L132C5-L133C62) and [Line 133](https://github.com/NRCan/geo-deep-learning/blob/develop/utils/metrics.py#L133) are swapped. In fact, Line 132 computes recall; line 133 computes precision, but they’re labeled the other way around.

**To Reproduce**
Consider a binary classification with the following ground truth and prediction labels:
`label_true = np.array([0, 0, 0, 1, 1, 1, 1, 1])`
`
label_pred = np.array([0, 1, 0, 1, 0, 1, 0, 1])
`
Then, the confusion matrix based on the function [calculate_confusion_matrix](https://github.com/NRCan/geo-deep-learning/blob/b0a2b73f72d129f6e8598907f887516ab176b989/utils/metrics.py#L62) is as follows:
`[[2 1],
 [2 3]]`
which corresponds to 
`[[TN FP],
[FN TP]]`
where TN, FP, FN, and TP are true negatives, false positives, false negatives, and true positives. Thus, for class 1 we should have the following precision and recall values:

- Precision = TP/(TP+FP) = 3/(3+1) = 0.75
- Recall = TP/(TP+FN) = 3/(3+2) = 0.6

Sklearn confirm the above values too. Try:
`precision_score(label_true, label_pred, pos_label=1, zero_division=0)`
`recall_score(label_true, label_pred, pos_label=1, zero_division=0)`

However, the code shows precision=0.60 and recall=0.75, they’re reversed.

**Comment by LucaRom** [comment]
Hi @FatemehEsfahani 

Thanks for reporting this.

The confusion matrix returned by `calculate_confusion_matrix` is laid out as [true, pred] (rows = ground truth, columns = predictions), which is consistent with `sklearn.metrics.confusion_matrix`.

But in the current implementation (lines 132–133), the two seem to be swapped as you said.

I think the fix would look like :

`precision = np.diag(batch_matrix) / batch_matrix.sum(axis=0)  # predictions
recall    = np.diag(batch_matrix) / batch_matrix.sum(axis=1)  # truths`

Feel free to submit a PR with the required changes, otherwise we will look into it at some point.

Just a note: we are in the process of releasing a major refactoring of GDL. In the new version, most metrics will be handled by third-party libraries such as TorchMetrics, so this kind of bug should be avoided in the future.
